src/schrodinger/rdkit_extensions/helm/cg_coordgen.cpp

```diff
@@ -679,11 +686,38 @@ static bool is_single_linear_polymer(const RDKit::ROMol& monomer_mol)
     return true;
 }
 
-static bool is_nucleic_acid(const RDKit::ROMol& polymer)
+static bool
+is_double_stranded_nucleic_acid(const RDKit::ROMol& monomer_mol,
+                                std::vector<RDKit::ROMOL_SPTR> polymers)
 {
-    auto polymerId = polymer.getProp<std::string>(POLYMER_ID);
-    return boost::starts_with(polymerId, "DNA") ||
-           boost::starts_with(polymerId, "RNA");
+    // return true if there's exactly two polymers, with no rings nor branches,
+    // and both are nucleic acids and there's at least one bond between them.
+    if (polymers.size() != 2) {
+        return false;
+    }
+    for (auto polymer : polymers) {
+        if (!is_nucleic_acid(*polymer)) {
+            return false;
+        }
+    }
+    const auto polymer_bonds =
+        get_bonds_between_polymers(*polymers[0], *polymers[1]);
+    if (polymer_bonds.size() < 1) {
+        return false;
+    }
+    compute_full_ring_info(monomer_mol);
+    if (monomer_mol.getRingInfo()->numRings() > 0) {
+        // reset ring info so it can get recomputed by the appropriate layout
+        // method
+        monomer_mol.getRingInfo()->reset();
+        return false;
+    }
+    for (auto monomer : monomer_mol.atoms()) {
+        if (monomer->getProp<bool>(BRANCH_MONOMER)) {
+            return false;
+        }
+    }
```

Comment:
I'm a little confused on this because I think double stranded NAs will definitely have branch monomers, I would expect this to always return false what am I missing? The base is always labeled as a branch monomer